### PR TITLE
More docu on standard bases

### PR DESCRIPTION
**Pull request by wdecker**
This is WIP as @ederc will add some more changes.

### CODE REVIEW

File: docs/src/CommutativeAlgebra/groebner_bases.md [hunk 1]
```diff
@@ -137,14 +141,14 @@ julia> tail(f)
...
-of the division process, the idea is to remove the leading term of the intermediate dividend using the leading term of some divisor
-by which it is divisible. Termination is guaranteed since $>$ is a well-ordering. In the more general case of standard bases we also
-allow local and mixed orderings. Here, a variant of the division algorithm due to Mora is used to guarantee termination.
+The computation of Gröbner (standard) bases relies on multivariate division with remainder which is interesting
+in its own right. If a monomial ordering $>$ is given, the basic idea is to mimic Euclidean division with remainder,
+allowing more than one divisor: At each step of the resulting process, this amounts to removing the leading term of the
+intermediate dividend, using the leading term of *some* divisor by which it is divisible. In its basic form, the process
+works well if $>$ is global, but may not terminate, however, for local and mixed orderings. In the latter case, Mora's division
+algorithm, which relies on a restricted selection strategy for the divisors to be used, comes to our rescue.
```
Comment (afkafkafk13) [inline docs/src/CommutativeAlgebra/groebner_bases.md:149]: Yes, this is short and very clear.

File: docs/src/CommutativeAlgebra/groebner_bases.md [hunk 2]
```diff
@@ -183,25 +187,21 @@ still sense to speak of fully reduced remainders. However, even if we start from
...
-
-```@julia
-reduce(f::T, F::Vector{T}; ordering::MonomialOrdering = default_ordering(parent(f))) where {T <: MPolyElem}
-```
-
-```@julia
-reduce(F::Vector{T}, G::Vector{T}; ordering::MonomialOrdering = default_ordering(parent(F[1]))) where {T <: MPolyElem}
+```@docs
+reduce(g::T, F::Vector{T}; 
+	ordering::MonomialOrdering = default_ordering(parent(f))) where {T <: MPolyElem}
```
Comment (afkafkafk13) [inline docs/src/CommutativeAlgebra/groebner_bases.md]: typo? Where does 'f' come from? Should this be F[1]?

File: docs/src/CommutativeAlgebra/groebner_bases.md [hunk 3]
```diff
@@ -211,38 +211,44 @@ Then the *leading ideal* of $G$ is the ideal of $K[x]$ defined by
...
+ordering is also called a *Gröbner basis*.
 
-We call a standard basis $G = \{g_1,\dots, g_r\}\subset K[x]_>\setminus \{0\}$ *minimal*  if  $\text{LM}_>(g_i)\neq \text{LM}_>(g_j)$
-for $i\neq j$.
+!!! note
+    Every standard basis of $I$ generates $I$.
 
 !!! note
-    The definition above deviates from the definition in most textbooks as we do not ask that the leading coefficients of the standard basis elements are 1.
+    Gröbner bases (standard bases) can be computed using Buchberger's algorithm (Buchberger's as algorithm enhanced by Mora).
```
Comment (afkafkafk13) [inline docs/src/CommutativeAlgebra/groebner_bases.md]: typo ? Additional 'as' in bracket

Comment (wdecker) [reply docs/src/CommutativeAlgebra/groebner_bases.md]: @ederc  Can you please change this to:   Buchberger's algorithm as enhanced by Mora

File: docs/src/CommutativeAlgebra/groebner_bases.md (hunk 2 above)
Comment (wdecker) [reply docs/src/CommutativeAlgebra/groebner_bases.md]: @ederc  Can you please replace f by F[1] both in groebner.jl and groebner_basis.md?

File: docs/src/CommutativeAlgebra/groebner_bases.md (hunk 3 above)
Comment (ederc) [reply docs/src/CommutativeAlgebra/groebner_bases.md]: Done

File: docs/src/CommutativeAlgebra/groebner_bases.md (hunk 2 above)
Comment (ederc) [reply docs/src/CommutativeAlgebra/groebner_bases.md]: Done

File: src/Rings/groebner.jl [hunk 4]
```diff
@@ -512,7 +563,70 @@ function reduce(F::Vector{T}, G::Vector{T}; ordering::MonomialOrdering = default
...
+# Examples
+```jldoctest
+julia> R, (x, y, z) = PolynomialRing(QQ, ["x", "y", "z"]);
+
+julia> f1 = x^2+x^2*y; f2 = y^3+x*y*z; f3 = x^3*y^2+z^4;
+
+julia> g = x^3*y+x^5+x^2*y^2*z^2+z^6;
+
+julia> u, Q, h = reduce_with_quotients_and_unit(g, [f1,f2, f3], ordering = negdegrevlex(R))
+([x^3-x*y^2*z^2+x*y+y^2*z^2 0 y*z^2+z^2], 0, [y+1])
```
Comment (afkafkafk13) [inline src/Rings/groebner.jl]: Please generate this result again. It is still in the old ordering of the output.

Comment (ederc) [reply src/Rings/groebner.jl]: Fixed with some other doctests.

**Comment by ederc** [comment]
For @wdecker and me this is good to go.

**Pull request merged**